src/rust/pal/linux/mod.rs

```diff
@@ -61,3 +62,15 @@ pub unsafe fn set_nonblock(fd: RawFd) -> i32 {
     flags |= libc::O_NONBLOCK;
     libc::fcntl(fd, libc::F_SETFL, flags, 1)
 }
+
+/// Converts a [std::net::SocketAddrV4] into a [libc::sockaddr_in].
+pub fn socketaddrv4_to_sockaddr_in(addr: &SocketAddrV4) -> libc::sockaddr_in {
+    libc::sockaddr_in {
+        sin_family: libc::AF_INET as u16,
+        sin_port: u16::to_be(addr.port()),
+        sin_addr: libc::in_addr {
+            s_addr: u32::to_be(u32::from_be_bytes(addr.ip().octets())),
+        },
```

Comment:
Again, libc uses its own type (`libc::in_addr_t`) as the type for `s_addr`.  So, I think line 72 should technically have "`as libc::in_addr_t`" added to the end of it.

But yes, this conversion is correct.  The version you're replacing was also correct (and avoided an unnecessary byte-swap), but only on little-endian machines.  The ideal thing here would probably be to define our own `pal::in_addr_t` type, with a from_be_bytes() method that creates the equivalent of a big-endian 32-bit unsigned value.

Reply:
Leaving off the casting was working, but I introduced it, so that we are technically correct.

Also, I have introduced conditional compilation to select the right byte conversion technique.